File: update_locales.py
import json
import os

locales_dir = r"c:\Users\newFUTURE\Desktop\xmax2\Savesnapfast\src\locales\locales"
en_path = os.path.join(locales_dir, "en.json")

# Keys to ensure exist in 'downloader'
required_keys = {
    "download_nwm": "Download No Watermark",
    "download_hd": "Download HD",
    "download_story_vid": "Download Story (Video)",
    "download_story_img": "Download Story (Image)"
}

# Read English file to get structure if needed, but we have hardcoded required keys for safety
try:
    with open(en_path, 'r', encoding='utf-8') as f:
        en_data = json.load(f)
        # optionally update required_keys from en_data if we wanted dynamic sync
except Exception as e:
    print(f"Error reading en.json: {e}")
    exit(1)

# Iterate over all json files
count = 0
for filename in os.listdir(locales_dir):
    if filename.endswith(".json") and filename != "en.json": # Skip en.json as it's the source
        file_path = os.path.join(locales_dir, filename)
        
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                data = json.load(f)
            
            if "downloader" not in data:
                data["downloader"] = {}
            
            modified = False
            for key, default_val in required_keys.items():
                if key not in data["downloader"]:
                    data["downloader"][key] = default_val # Use English default
                    modified = True
                    print(f"Added {key} to {filename}")
            
            if modified:
                with open(file_path, 'w', encoding='utf-8') as f:
                    json.dump(data, f, ensure_ascii=False, indent=2)
                count += 1
                
        except Exception as e:
            print(f"Error processing {filename}: {e}")

print(f"Updated {count} files.")

File: update_locales.js

const fs = require('fs');
const path = require('path');

const localesDir = path.join(__dirname, 'src', 'locales', 'locales');
const enPath = path.join(localesDir, 'en.json');

// Keys to ensure exist in 'downloader'
const requiredKeys = {
    "download_nwm": "Download No Watermark",
    "download_hd": "Download HD",
    "download_story_vid": "Download Story (Video)",
    "download_story_img": "Download Story (Image)"
};

try {
    const files = fs.readdirSync(localesDir);
    let count = 0;

    files.forEach(filename => {
        if (filename.endsWith('.json') && filename !== 'en.json') {
            const filePath = path.join(localesDir, filename);

            try {
                const content = fs.readFileSync(filePath, 'utf8');
                const data = JSON.parse(content);

                if (!data.downloader) {
                    data.downloader = {};
                }

                let modified = false;
                for (const [key, defaultVal] of Object.entries(requiredKeys)) {
                    if (!data.downloader[key]) {
                        // Use English default if missing
                        data.downloader[key] = defaultVal;
                        modified = true;
                    }
                }

                if (modified) {
                    fs.writeFileSync(filePath, JSON.stringify(data, null, 2), 'utf8');
                    console.log(`Updated ${filename}`);
                    count++;
                }
            } catch (err) {
                console.error(`Error processing ${filename}:`, err.message);
            }
        }
    });

    console.log(`Successfully updated ${count} files.`);

} catch (err) {
    console.error("Error reading dir:", err);
}
